unstructured/ingest/connector/confluence.py

```diff
@@ -0,0 +1,208 @@
+import math
+import os
+from dataclasses import dataclass
+from pathlib import Path
+
+from atlassian import Confluence
+
+from unstructured.ingest.interfaces import (
+    BaseConnector,
+    BaseConnectorConfig,
+    BaseIngestDoc,
+    ConnectorCleanupMixin,
+    IngestDocCleanupMixin,
+    StandardConnectorConfig,
+)
+from unstructured.ingest.logger import logger
+from unstructured.utils import requires_dependencies
+
+
+@dataclass
+class SimpleConfluenceConfig(BaseConnectorConfig):
+    """Connector config where:
+    user_email is the email to authenticate into Confluence Cloud,
+    api_token is the api token to authenticate into Confluence Cloud,
+    and url is the URL pointing to the Confluence Cloud instance.
+
+    Check https://developer.atlassian.com/cloud/confluence/basic-auth-for-rest-apis/
+    for more info on the api_token.
+    """
+
+    user_email: str
+    api_token: str
+    url: str
+    list_of_spaces: str
+    max_number_of_spaces: int
+    max_number_of_docs_from_each_space: int
+
+
+@dataclass
+class ConfluenceFileMeta:
+    """Metadata specifying:
+    id for the confluence space that the document locates in,
+    and the id of document that is being reached to.
+    """
+
+    space_id: str
+    document_id: str
+
+
+def scroll_wrapper(func):
+    def wrapper(*args, **kwargs):
+        """Wraps a function to obtain scroll functionality."""
+        number_of_items_to_fetch = kwargs["number_of_items_to_fetch"]
+        del kwargs["number_of_items_to_fetch"]
+
+        kwargs["limit"] = min(100, number_of_items_to_fetch)
+        kwargs["start"] = 0 if "start" not in kwargs else kwargs["start"]
+
+        all_results = []
+        num_iterations = math.ceil(number_of_items_to_fetch / kwargs["limit"])
+
+        for _ in range(num_iterations):
+            response = func(*args, **kwargs)
+            if type(response) is list:
+                all_results += func(*args, **kwargs)
+            elif type(response) is dict:
+                all_results += func(*args, **kwargs)["results"]
+
+            kwargs["start"] += kwargs["limit"]
+
+        return all_results[:number_of_items_to_fetch]
+
+    return wrapper
+
+
+@dataclass
+class ConfluenceIngestDoc(IngestDocCleanupMixin, BaseIngestDoc):
+    """Class encapsulating fetching a doc and writing processed results (but not
+    doing the processing!).
+
+    Current implementation creates a Confluence connection object
+    to fetch each doc, rather than creating a it for each thread.
+    """
+
+    config: SimpleConfluenceConfig
+    file_meta: ConfluenceFileMeta
+
+    # TODO: remove one of filename or _tmp_download_file, using a wrapper
+    @property
+    def filename(self):
+        return (
+            Path(self.standard_config.download_dir)
+            / self.file_meta.space_id
+            / f"{self.file_meta.document_id}.txt"
```

Comment:
does this always return valid html? maybe we just make this a .html file so we don't miss the on the auto detection?

Reply:
File naming might indeed be the reason for the above problem. I'll check if it always returns html, and if so I'll change this part 👍

Reply:
Checking from [atlassian rest client](https://github.com/atlassian-api/atlassian-python-api/blob/5d9cd15a91eea0384333d408ae0e68148c7dc302/atlassian/rest_client.py#L285), which gets called from [a method within the confluence module](https://github.com/atlassian-api/atlassian-python-api/blob/5d9cd15a91eea0384333d408ae0e68148c7dc302/atlassian/confluence.py#L320)

- Rest client makes a request to the Confluence page using `requests`
- tries to decode the response as json, 
- and if it can't decode as json, it returns `response.text `

I think it is safe to say confluence pages return html responses, given our results; which means `response.text` would be html. I've also checked responses from multiple websites around the web to verify this. (in addition to the verification from our ingest download results)

JSON responses should be for other types of content (such as content served on API endpoints).

Reply:
beautiful!